vivisect/analysis/i386/golang.py

```diff
@@ -0,0 +1,202 @@
+    ep = vw.getEntryPoints()
+    if len(ep) != 1:
+        return
+
+    # Search from the entry point function.
+    ep_va = ep[0]
+    bblocks = vw.getFunctionBlocks(ep_va)
+    if not bblocks:
+        return
+    ptr_va, runtime_va = extract_golang_mainmain(vw, bblocks, filename)
+    if runtime_va is None:
+        return
+
+    # Invoke codeflow on runtime_main().
+    vw.makeFunction(runtime_va)
+
+    # Also mark the ptr_va as a pointer to runtime_va.
+    vw.makePointer(ptr_va, tova=runtime_va)
+
+    return
+
+
+def extract_golang_mainmain(vw, basic_blocks, filename):
+    '''
+    Find the basic block of interest and return the address of
+    the pointer and its contents, runtime_main().
+
+    The BB will contain the sequence of opcodes in _GOLANG_I386_INSTRS.
+    The push instruction -5 from the end will load the contents of a memory
+    address, and those contents are the runtime_main() address.
+    '''
+    op = find_golang_bblock_1(vw, basic_blocks, filename)
+    if op is None:
+        op = find_golang_bblock_2(vw, filename)
+        if op is None:
+            return None, None
+
+    # The key opcode is "push immediate", which identifies a pointer which
+    # in turn identifies the GO function runtime_mainPC (aka runtime_main).
+    ptr_va, runtime_va = parse_push_imm(vw, op, filename, get_content=True)
+    return ptr_va, runtime_va
+
+
+# Opcode sequence, where element [-5] is the important one.
+_GOLANG_I386_INSTRS = ['cld', 'call', 'mov', 'mov', 'mov', 'mov', 'call',
+                       'call', 'call', 'push', 'push', 'call', 'pop', 'pop']
+
+
+def find_golang_bblock_1(vw, basic_blocks, filename):
+    '''
+    Find the basic block of interest and return the opcode where
+    the special sequence of opcodes begins.  Return None if not found.
+
+    For this case, Vivisect interprets the entry as one large function
+    (IDA Pro may break it into 2 or 3 functions linked by jmp instructions).
+    The basic block of interest is in this function.
+    '''
+    the_bblock = None
+    for bblock in basic_blocks:
+        next_va = bblock[0]
+        instrs = []
+        while next_va < bblock[0] + bblock[1]:
+            try:
+                op = vw.parseOpcode(next_va)
+            except envi.InvalidInstruction:
+                op = None
+            if op is None:
+                return None
+            if (len(instrs) > 0) or (op.mnem == _GOLANG_I386_INSTRS[0]):
+                # Record instructions from the first opcode match.
+                instrs.append(op)
+            next_va += op.size
+        if len(instrs) >= len(_GOLANG_I386_INSTRS):
+            the_bblock = bblock
+            break
+    if the_bblock is None:
+        return None
+
+    for k in range(len(_GOLANG_I386_INSTRS)):
+        if instrs[k].mnem != _GOLANG_I386_INSTRS[k]:
+            return None
+    return instrs[len(_GOLANG_I386_INSTRS) - 5]
+
+
+def find_golang_bblock_2(vw, filename):
```

Comment:
So, since `find_golang_bblock_2` also calls `find_golang_bblock_` (but `_1` is called before it in the , is there a more descriptive name for the two we can find?